js/app.js

```diff
@@ -0,0 +1,159 @@
+                    break;
+                case 'w':
+                    this.carData.carPos.y -= 25
+                    break;
+                case 'ArrowDown':
+                    this.carData.carPos.y += 25
+                    break;
+                case 's':
+                    this.carData.carPos.y += 25
+                    break;
+            }
+        }
+    },
+    drawSquare() {
+        this.ctx.fillStyle = '#C8DBBE'
+        this.ctx.fillRect(0, 0, this.canvasSize.w, this.canvasSize.h)
+        this.ctx.fillStyle = '#B7C4CF'
+        this.ctx.fillRect(50, 0, this.canvasSize.w - 100, this.canvasSize.h)
+    },
+    drawLines() {
+        this.ctx.strokeStyle = '#FDFDBD'
+        this.ctx.lineWidth = 15
+        this.ctx.beginPath()
+        this.ctx.moveTo(70, 0)
+        this.ctx.lineTo(70, this.canvasSize.h)
+        this.ctx.stroke()
+        this.ctx.closePath()
+
+        this.ctx.beginPath()
+        this.ctx.moveTo(this.canvasSize.w - 70, 0)
+        this.ctx.lineTo(this.canvasSize.w - 70, this.canvasSize.h)
+        this.ctx.stroke()
+        this.ctx.closePath()
+
+        this.ctx.strokeStyle = '#FDFDBD'
+        this.ctx.lineWidth = 10
+        this.ctx.beginPath()
+        this.ctx.setLineDash([70, 20])
+        this.ctx.moveTo(this.canvasSize.w / 2, 0)
+        this.ctx.lineTo(this.canvasSize.w / 2, this.canvasSize.h)
+        this.ctx.stroke()
+        this.ctx.closePath()
+        this.ctx.setLineDash([0, 0])
+    },
+    drawRoad() {
+        this.drawSquare()
+        this.drawLines()
+    },
+    createCar() {
+        this.imageInstance = new Image()
+        this.imageInstance.src = this.carData.image
+    },
+    start() {
+        setInterval(() => {
+            this.framesCounter++
+            if (this.framesCounter % 35 === 0) {
+                this.createObstacle()
+                this.scoreCouter++
+            }
+            this.clearAll()
+            this.moveAll()
+            this.drawAll()
+            this.checkCollision()
+        }, 50)
+    },
+    clearAll() {
+        this.ctx.clearRect(0, 0, this.canvasSize.w, this.canvasSize.h)
+    },
+    moveAll() {
+        this.obstacle.forEach(elem => elem.move())
+    },
+    drawAll() {
+        this.drawRoad()
+        this.ctx.drawImage(
+            this.imageInstance,
+            this.carData.carPos.x,
+            this.carData.carPos.y,
+            this.carData.carSize.w,
+            this.carData.carSize.h)
+        this.obstacle.forEach(elem => elem.draw())
+    },
+    checkCollision() {
+        this.obstacle.forEach(elem => {
+            if (elem.obstaclePos.x < this.carData.carPos.x + this.carData.carSize.w &&
+                elem.obstaclePos.x + elem.obstacleSize.w > this.carData.carPos.x &&
+                elem.obstaclePos.y < this.carData.carPos.y + this.carData.carSize.h &&
+                elem.obstacleSize.h + elem.obstaclePos.y > this.carData.carPos.y) {
+                // Collision
+                clearInterval(1)
```

Comment:
Estaría chulo que te hubieses creado una variable para guardar el setInterval y luego utilizarla para el clearInterval.
clearInterval(this.intervalId)